### DISCUSSION TITLE
"Error: no suitable executable found" for executable in $HOME

### DISCUSSION THREAD

**Discussion by tredondo** [netblue30/firejail#4130] [Q&A] [answered]
I've downloaded an Electron-based video editing app ([LosslessCut](https://github.com/mifi/lossless-cut/releases/latest/download/LosslessCut-linux.tar.bz2)) and I'm trying to run it with existing profiles from video players, but I get "Error: no suitable executable found", even though the executable exists and has the `+x` attribute set. If I don't pass a profile to `firejail`, it doesn't say that no suitable executable was found.

```sh
$ firejail --profile=vlc ./losslesscut
Reading profile /etc/firejail/vlc.profile
Reading profile /etc/firejail/disable-common.inc
[...]
Parent pid 3506228, child pid 3506229
21 programs installed in 40.69 ms
Warning: cleaning all supplementary groups
Child process initialized in 154.33 ms
Error: no suitable ./losslesscut executable found

Parent is shutting down, bye...

$ firejail ./losslesscut                    
Reading profile /etc/firejail/default.profile
[...]
** Note: you can use --noprofile to disable default.profile **

Parent pid 3506191, child pid 3506192
Warning: cleaning all supplementary groups
Child process initialized in 59.45 ms

Parent is shutting down, bye...

$ file ./losslesscut 
./losslesscut: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=cbfaeba665f2fd7acdfb55da58780d895731df17, stripped
```

**Comment by rusty-snake** [comment, marked as answer]
> I've try to run a program with a existing profile of an other program. *[formulated by me]*

That's a bad idea and can not work well. Leave it and write a own profile for it (maybe based on that other) or use the default profile (with some extra arguments).

> even though the executable exists and has the +x attribute set

Outside of the sandbox yes, but also inside the sandbox? (rhetorical question)
You can check this with `firejail --ignore=private-bin --profile=PROFILE ls -l ./losslesscut`

> "Error: no suitable executable found" (and alike)

The most common reasons for this error:
 1. The executable (or a upper directory) is `blacklist`ed.
 2. The executable is under a `whitelist`ed directory but is not `whitelist`.
 3. The executable is missing in `private-bin`.
 4. The executable is under a `noexec` directory.

vlc.profile is a `whitelist`ing profile that `include`s `disable-exec.inc` and your program seems to be at `~/losslesscut` so you need

```
ignore noexec ${HOME}
whitelist ${HOME}/losslesscut
```

> an Electron-based app

Use `electron.profile` or start it with `--no-sandbox`, otherwise that firejail sandbox and the chromium sandbox conflict and you won't be able to start the app.
